src/libs/actions/Transaction.ts

```diff
@@ -56,10 +57,10 @@ function addStop(transactionID: string) {
     });
 }
 
-function saveWaypoint(transactionID: string, index: string, waypoint: RecentWaypoint | null, isEditingWaypoint = false) {
-    Onyx.merge(`${ONYXKEYS.COLLECTION.TRANSACTION}${transactionID}`, {
+function saveWaypoint(transactionID: string, index: string, waypoint: RecentWaypoint | null, isDraft = false) {
+    Onyx.merge(`${isDraft ? ONYXKEYS.COLLECTION.TRANSACTION : ONYXKEYS.COLLECTION.TRANSACTION_DRAFT}${transactionID}`, {
```

Comment:
@ntdiary I think it makes sense because isDraft = isEditingWaypoint, we will update the pending field to UPDATE if isEditingWaypoint is true. If isDraft == isEditingWaypoint == false It means that we are creating new waypoint so we update the pending field to ADD

But my curious here that why we need to update the pending field here even though we don't call API

Reply:
Ugh, you're right 🤦 I got them backwards with the logic.